legend-engine-xt-relationalStore-spanner-jdbc-shaded/pom.xml

```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>legend-engine</artifactId>
+        <groupId>org.finos.legend.engine</groupId>
+        <version>3.28.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>legend-engine-xt-relationalStore-spanner-jdbc-shaded</artifactId>
+    <name>Legend Engine - XT - Relational Store - Spanner - JDBC Driver - Shaded</name>
+
+    <properties>
+        <maven-shade-plugin.shadedPattern>org.finos.legend.engine.spanner.jdbc.shaded</maven-shade-plugin.shadedPattern>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.cloud</groupId>
+            <artifactId>google-cloud-spanner-jdbc</artifactId>
+            <version>2.7.3</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-shade-plugin</artifactId>
+                <configuration>
```

Comment:
After shading and relocation shade plugin creates dependency-reduced.pom with no dependencies and shaded.jar
To use it I triggered install plugin with -local postfix.
<classifier> won't work cause classifiers doesn't remove dependencies.

At the end of the day there is no exclusions now

Reply:
@rafaelbey 
It seems not possible due to fact that shaded library is a submodule, not separate project.
`mvn install legend-engine-xt-relationalStore-spanner-jdbc-shaded` is creating in local repo correct jar and correct pom **without** any deps.

Nevertheless, using this submodule in other submodules ends with an error. I think that this is because maven reads pom inside legend-engine-xt-relationalStore-spanner-jdbc-shaded, not using it from local repo.

But, if legend-engine-xt-relationalStore-spanner-jdbc-shaded would be moved to separate project, exclusion can be omitted.
Take a look at picture of maven with submodule being part of the project:
 
![image](https://user-images.githubusercontent.com/15000589/206484133-07b41c7a-a025-455d-a010-0332ebf900ff.png)

and now if it is separate:
![image](https://user-images.githubusercontent.com/15000589/206485816-92f19918-c899-45a1-a26d-2ad25fb434e2.png)

Note than in first case I can expand deps, and in second not. 
I did build with external project without exclusions and it went well.

Also, in Alloy now jdbc submodule is used without exclusion that proves that having shaded lib in separate project is crutual.

Reply:
It seems like maven enforcer plugin uses pom.xml instead of dependency-reduced-pom.xml, when the dependency is a module within the project. This requires us to put * exclusion to satisfy enforcer.

However, @roasher can you try adding https://maven.apache.org/plugins/maven-shade-plugin/shade-mojo.html#shadedArtifactAttached to the shade-plugin, and then add <classifier>shaded</classifier> to the dependency declarations. Maybe, enforcer plugin will handle shaded classifier as a special case, and not complain.

If it still complains, then using * exclusion with a comment explaining why, is better.

Reply:
@vikask-gs 
Checked that - didn't work.
if shadedClassifierName option is used, one need to set shadedArtifactAttached to true as well: 
```
<shadedArtifactAttached>true</shadedArtifactAttached>
<shadedClassifierName>shaded</shadedClassifierName>
```
So, it that case we are telling maven to create additional version with "shaded" classifier, but the problem is that we are not replacing output jar and pom with shaded one: there 2 jars in outcome and single pom. And that pom as long as it's not replaced with dependency-reduced.pom would still have google-cloud-spanner-jdbc dependency that would force to exclude everything not only in legend-engine, but in all projects that are using is as a dependency.

Reply:
We are not replacing the pom. But I was hoping that if you add the dependency with `<classifier>shaded</classifier>`, then enforcer plugin might automatically exclude transitive deps from pom.

Can you confirm again that it is not the case? 

If yes, then using * exclusion with a comment is fine.

Reply:
Yes, I can.
Added comment for exclusion.